lib/engine/game/base.rb

```diff
@@ -849,19 +852,43 @@ def next_round!
         @round_history << @actions.size
       end
 
+      def game_end_check
+        when_points = [:immediate]
+        when_points << :current_or if @round.is_a?(Round::Operating)
+        when_points << :full_or if @round.is_a?(Round::Operating) && @round.round_num == @operating_rounds
+
+        triggers = {
+          bankrupt: -> { bankruptcy_limit_reached? },
+          bank: -> { @bank.broken? },
+          stockmarket: -> { @stock_market.max_reached? },
```

Comment:
there's no need to make these lambdas, just leave them booleans

Reply:
Fixed next push